tools/ddprobe.cpp

```diff
@@ -92,24 +214,41 @@ test_dxgi_duplication(dxgi::adapter_t &adapter, dxgi::output_t &output) {
     &device,
     nullptr,
     nullptr);
+
   if (FAILED(status)) {
-    std::cout << "Failed to create D3D11 device for DD test [0x"sv << util::hex(status).to_string_view() << ']' << std::endl;
+    std::cout << "Failed to create D3D11 device for DD test [0x" << std::hex << status << "]" << std::endl;
     return status;
   }
 
   dxgi::output1_t output1;
   status = output->QueryInterface(IID_IDXGIOutput1, (void **) &output1);
   if (FAILED(status)) {
-    std::cout << "Failed to query IDXGIOutput1 from the output"sv << std::endl;
+    std::cout << "Failed to query IDXGIOutput1 from the output [0x" << std::hex << status << "]" << std::endl;
     return status;
   }
 
   // Ensure we can duplicate the current display
   syncThreadDesktop();
 
-  // Return the result of DuplicateOutput() to Sunshine
+  // Attempt to duplicate the output
   dxgi::dup_t dup;
-  return output1->DuplicateOutput((IUnknown *) device.get(), &dup);
+  HRESULT result = output1->DuplicateOutput(static_cast<IUnknown *>(device.get()), &dup);
+
+  if (SUCCEEDED(result)) {
+    // If duplication is successful, test frame capture
+    HRESULT captureResult = test_frame_capture(dup, device.get());
+    if (SUCCEEDED(captureResult)) {
+      return S_OK;
+    }
+    else {
+      std::cout << "Frame capture test failed [0x" << std::hex << captureResult << "]" << std::endl;
+      return captureResult;
+    }
+  }
+  else {
```

Comment:
Redundant else.

Reply:
Same as before, this is more a programmers preference. A lot of devs include else for clarity even if it doesn't matter.